### ISSUE TITLE
Unexpected behaviour with random.shuffle(tensor)

### ISSUE THREAD

**Issue by fatchord** [pytorch/pytorch#18934]
As you can see it's not a true shuffle as there are repeats and missing values.

>>> import torch, random
>>> import numpy as np
>>> x1 = torch.arange(10)
>>> x2 = np.arange(10)
>>> x1, x2
(tensor([0, 1, 2, 3, 4, 5, 6, 7, 8, 9]), array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9]))
>>> random.shuffle(x1)
>>> random.shuffle(x2)
>>> x1, x2
(tensor([0, 0, 2, 0, 2, 2, 6, 1, 4, 4]), array([1, 3, 8, 0, 2, 6, 5, 4, 7, 9]))

**Comment by vishwakftw** [comment]
`random.shuffle` shuffles the contents of the Tensor in-place, using element swapping in a Pythonic way.

Element swapping doesn't work for Tensors as reported in this issue: https://github.com/pytorch/pytorch/issues/17008

**Comment by umanwizard** [comment]
Appears to be a duplicate of #17008 .
